api/src/main/java/org/openmrs/module/fhir/api/util/FHIRDiagnosticReportUtil.java

```diff
@@ -13,46 +13,173 @@
+
+		// Set parsed obsSet (`Result` as Set of Obs)
+		// Set Binary Obs Handler which used to store `PresentedForm`
+
+		return omrsDiagnosticReport;
+	}
+
+	/**
+	 * Update a OpenMRS Diagnostic Report (Encounter) with a given FHIR
+	 * Diagnostic Report
+	 * Note: Since this method is updating a OpenMRS Diagnostic Report,
+	 * it'll only calling to UPDATE method in given Handler
+	 *
+	 * @param diagnosticReport     FHIR Diagnostic Report
+	 * @param omrsDiagnosticReport OpenMRS Diagnostic Report (Encounter)
+	 * @param handler              An implementation of DiagnosticReportHandler
+	 * @return An instance of org.openmrs.Encounter
+	 */
+	public static Encounter updateOpenMRSDiagnosticReport(DiagnosticReport diagnosticReport, Encounter omrsDiagnosticReport,
+	                                                      DiagnosticReportHandler handler) {
+		// Update `Name` as a Obs
+		// Update `Status` as a Obs
+		// Update `Issued` date as EncounterDateTime
+		// Update `Subject` as Encounter Patient
+		// Update `Performer` as Encounter Provider
+		// Update `ServiceCategory` as EncounterType
+		// Update `Diagnosis[x]->DateTime` as valueDateTime in an Obs
+		// Update `Diagnosis[x]->Period` as valueDateTime in an Obs
+
+		// Update parsed obsSet (`Result` as Set of Obs)
+		// Update Binary Obs Handler which used to store `PresentedForm`
+
+		return omrsDiagnosticReport;
+	}
+
+	/**
+	 * Delete a OpenMRS Diagnostic Report (Encounter)
+	 * Note: It'll only calling to DELETE method in give Handler
+	 *
+	 * @param omrsDiagnosticReport FHIR Diagnostic Report
+	 * @param handler              An implementation of DiagnosticReportHandler
+	 * @return An instance of org.openmrs.Encounter
+	 */
+	public static Encounter deleteOpenMRSDiagnosticReport(Encounter omrsDiagnosticReport, DiagnosticReportHandler
+			handler) {
+		// Delete `Name` Obs
+		// Delete `Status` Obs
+
+		// Delete Obs (`Result` as Set of Obs)
+		// Delete Binary Obs Handler which used to store `PresentedForm`
+
+		return omrsDiagnosticReport;
+	}
+
+	public static List<Obs> getOpenMRSObs(DiagnosticReport diagnosticReport) {
+		List<Obs> obsSet = new ArrayList<Obs>();
+		for (ResourceReferenceDt reference : diagnosticReport.getResult()) {
+			if (reference.getReference().isLocal()) {
+				Observation obs = (Observation) reference.getResource();
+				// obsSet.add(FHIRObsUtil.generateOpenMRSObs(obs, new ArrayList<String>()));
+			} else {
+				getOpenMRSObs(reference);
+			}
+		}
+		return obsSet;
+	}
+
+	private static Obs getOpenMRSObs(ResourceReferenceDt reference) {
+		if (reference.getReference().isLocal()) {
+			Observation obs = (Observation) reference.getResource();
+			return FHIRObsUtil.generateOpenMRSObs(obs, new ArrayList<String>());
+		} else {
+			return getExternalObsResource(reference);
+		}
+	}
+
+	private static Obs getExternalObsResource(ResourceReferenceDt reference) {
+		log.warn("Doesn't implement fetch external resources.", new NotImplementedException());
```

Comment:
@harsha89 I have no idea about what you said here. How can I correct this ? What are the log levels? 

This method is fetching external reference resources. We are not considering about these operations at this level of implementation. So, this method should not get executed.